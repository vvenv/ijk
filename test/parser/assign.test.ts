import { expect, test } from 'vitest';
import { parse } from './__helper';

test('basic', () => {
  expect(parse(`{{ assign x = y }}{{ x }} = {{ y }}`)).toMatchSnapshot();
});

test('destructuring', () => {
  expect(
    parse(`{{ assign x,y,z = a }}{{ x }},{{ y }},{{ z }}`),
  ).toMatchSnapshot();
});

test('duo', () => {
  expect(
    parse(`{{ assign x = 'y' }}{{ x }}{{ assign x = "y" }}{{ x }}`),
  ).toMatchSnapshot();
});

test('nesting', () => {
  expect(
    parse(`{{ assign x = y }}{{ assign z = x }}{{ x }} = {{ y }} = {{ z }}`),
  ).toMatchSnapshot();
});

test('filter', () => {
  expect(
    parse(`{{ assign x = y | split }}{{ x }} from {{ y }}`),
  ).toMatchSnapshot();
});

test('filter w/ params', () => {
  expect(
    parse(`{{ assign char = name | split "" }}{{ char }} from {{ name }}`),
  ).toMatchSnapshot();
});
