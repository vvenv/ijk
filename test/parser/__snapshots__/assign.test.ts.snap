// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`basic 1`] = `""use strict";let s="";Object.assign(c,{x:c.y,});s+=e(c.x);s+=" = ";s+=e(c.y);return s;"`;

exports[`destructuring 1`] = `""use strict";let s="";Object.assign(c,{x:c.a.x,y:c.a.y,z:c.a.z,});s+=e(c.x);s+=",";s+=e(c.y);s+=",";s+=e(c.z);return s;"`;

exports[`duo 1`] = `""use strict";let s="";Object.assign(c,{x:'y',});s+=e(c.x);Object.assign(c,{x:"y",});s+=e(c.x);return s;"`;

exports[`filter 1`] = `""use strict";let s="";Object.assign(c,{x:f.split(c.y),});s+=e(c.x);s+=" from ";s+=e(c.y);return s;"`;

exports[`filter w/ params 1`] = `""use strict";let s="";Object.assign(c,{char:f.split(c.name,""),});s+=e(c.char);s+=" from ";s+=e(c.name);return s;"`;

exports[`nesting 1`] = `""use strict";let s="";Object.assign(c,{x:c.y,});Object.assign(c,{z:c.x,});s+=e(c.x);s+=" = ";s+=e(c.y);s+=" = ";s+=e(c.z);return s;"`;
